MigrationScript/Migration.ps1

```diff
@@ -144,7 +182,194 @@ function Main
     $logger.LogInfoObject("Migrated AMSv1 SapNeWeaver List - ", $sapNetWeaverTransformedList)
     $logger.LogInfoObject("Not Migrated AMSv1 Unsupported Provider list - ", $unsupportedProviderList)
 
+	$logger.LogInfo("-----------Finishing migration to AMSv2--------------")
+
     Stop-Transcript
 }
 
+<#
+.SYNOPSIS
+Function to migrate the Hana Providers to ams v2
+
+.PARAMETER secretName
+Secret Name, this contains the name of the provider. 
+
+.PARAMETER secretValue
+Secret Value, this contains the provider configuration.
+
+.PARAMETER logger
+logger object.
+
+.EXAMPLE
+MigrateHanaProvider -secretName $secret.name -secretValue $secret -logger $logger
+#>
+function MigrateHanaProvider([string]$secretName, $secretValue, $logger) {
+	$parsedArmId = Get-ParsedArmId $amsv2ArmId
+
+	[string]$monitorName = $parsedArmId.amsResourceName;
+	[string]$resourceGroupName = $parsedArmId.amsResourceGroup;
+	[string]$subscriptionId = $parsedArmId.subscriptionId;
+	# set context.
+	Set-AzContext -SubscriptionId $subscriptionId -TenantId $tenantId;
+
+	[string]$providerType = $($secretValue.type);
+	[string]$providerName = $($secretValue.name) # + (Get-Random -Minimum 2 -Maximum 20000).ToString();
```

Comment:
Remove this comment, also lets hardcode the provider name here, since the input is also case sensitive.

Reply:
Removed the comment and added constants for providerType